README.md
archivczsk
==========
enigma2 plugin pre prehravanie cz/sk video archivov

**Info:** https://github.com/mx3L/archivczsk/releases

**Doplnky:** https://github.com/mx3L/archivczsk-doplnky


CHANGELOG.md
## Zoznam zmien: 

  * automaticka aktualizacia pluginu so zachovanim udajov
  * volba CSFD pluginu (+interny)
  * rozsirene informacie o filme (pokial ich doplnok poskytuje)
  * definovanie vlastneho radenia doplnok (prinosne pri vlastnej kategorii)
  * predny panel setoboxu zobrazuje co sa prave prehrava
  * logovanie (default: /tmp/archivCZSK.log)
  * vylepsena funkcionalita aktualizacie doplnkov
  * kontrola aktualizacii pri spusteni pluginu resp. po uplynuti 2h od poslednej kontroly
  * uprava designu skinov
  * rychlostne vylepsenie pluginu
  * fix FullHD skinu ktory sposoboval pad systemu
  * fix virtualnej klavesnice (XcursorX)
  * fix pridavania do oblubenych poloziek (niekedy to nic nepridalo)
  * fix pretacania v doplnku Sosac (pri zaplatenom VIP ucte)
  * fix kompatibility OE2.0 - OE2.5
  * fix chybajuceho kodovania napr. v OpenAtv6.1 (pad systemu pri stahovani videa)
  * fix dalsich chyb ktore sposobovali pad systemu


## Install
Pouzivatelom windows doporucujem stiahnut program[ __putty__](https://the.earth.li/~sgtatham/putty/latest/w32/putty.exe) cez ktory sa pripojite na setobox samozrejme musite poznat IP adresu setoboxu a prihlasovacie meno zvycajne *__root__*

* __SSH telnet__

      opkg install curl
      opkg install http://github.com/mx3L/archivczsk/releases/download/v1.0/subssupport_1.5.5-20170116_all.ipk
      opkg install http://github.com/.............ipk
      init 4
      init 3
      (alebo: reboot)
      
      -----------------------Debian-----------------------

      dpkg -i curl
      dpkg -i http://github.com/mx3L/archivczsk/releases/download/v1.0/subssupport_1.5.5-20170116_all.deb
      dpkg -i http://github.com/.............deb
      init 4
      init 3
      (alebo: reboot)
      
* __Manual__ (nedoporucujem, mozne problemy)
   * stiahnut balicek (*.ipk|*.deb)  a nahrat do setoboxu do adresara __/tmp__
   * cez software setoboxu najst manualne instalovanie balickov a mal by sa objavit dany nazov balicku
   * naistalovat a restart setoboxu
   * *nebude fungovat auto update pokial nie je naistalovany __curl__*

## Re-Install
Ak uz mate jednu z predoslych verzii naistalovanu obnovia sa kategorie aj data pluginov ktore ste mali pred tym.

* __SSH telnet__

      mkdir -p /tmp/archivczsk
      cp -R /usr/lib/enigma2/python/Plugins/Extensions/archivCZSK/resources/data /tmp/archivczsk
      opkg remove -force-depends enigma2-plugin-extensions-archivczsk
      opkg install http://github.com/.............ipk
      init 4
      init 3
            
      -----------------------Debian-----------------------

      mkdir -p /tmp/archivczsk
      cp -R /usr/lib/enigma2/python/Plugins/Extensions/archivCZSK/resources/data /tmp/archivczsk
      dpkg --purge enigma2-plugin-extensions-archivczsk
      dpkg -i http://github.com/.............deb
      init 4
      init 3
      (alebo: reboot)

## Problem spustenia pluginu (napr. pre Black Hole image)
- Pokial sa Vam nezobrazi archivCZSK v ponuke resp. pri vstupe do zoznamu modulov Vam vypise "cannot import name BrowserLikeRedirectAgent" tak treba spustit cez telnet tento prikaz a restartnut prijmac (fix zaberie cca +4MB v pamati FLASH ale kto chce moze vymazat zolzku '/twisted_origin_backup' a subory *.py v zlozke '/twisted' __az po restarte prijmaca!!!__). 

  ```
  opkg install http://github.com/mx3L/archivczsk/releases/download/v1.0/twisted-16-2-0-fix.ipk
  ```

## Reporting
Pripadne chyby poprosim hlasit sem na github (__doporucujem pripojit log file__ inac je dost mozne ze nedostanete odpoved)





*********************************************************************************************************
*********************************************************************************************************




## Verzia 0.73
**Vseobecne**

* pridana full-hd verzia skinu
* pridana podpora pre skinovanie nastaveni doplnkov/archivu
  * uprava skinu nastaveni z vyuzitim noveho Tabs komponentu
* opravy a vylepsenia pri zobrazovani cesty pri prehliadani doplnkov
  * vzdy sa zobrazuje spravna cesta
  * skracovanie cesty ak za sebou nasleduju rovnake nazvy 
* pridane vyhladavanie v EPG pre sosac plugin
* pridana podpora pre editovanie ulozenych vyhladavanych nazvov
* oprava hlasenej chyby SSL: CERTIFICATE_VERIFY_FAILED, v niektorych 
doplnkoch(ivysilani, prima, koukni) pri pouziti python-2.7.9+
* oprava stahovania aktualizacii po prvom spusteni
* odstranene nepotrebne sucasti archivov (servicemp4, rtmpgw)
* odstranene instalovanie rtmpdump z dovodu moznej nekompatibility
a uprednostnene instalovanie rtmpdump z feedu

## Verzia 0.72 beta 8
**Vseobecne**

* oprava GS pri vyhladavani cez EPG 
* pridana podpora pre novy prima plugin
* odstraneny nefunkcny prima plugin,
* oprava tv.sme.sk a videacesky.cz 

## Verzia 0.72 beta 7
**Vseobecne**

* aktualizovane archivy
* mensie upravy skinu, priprava pre podporu inych skinov
* oprava GS pred zacatim prehravania na OpenATV image
* odstraneny prehravac streamov ( odporucam pouzit IPTV bouquets )
* pridana moznost neotvarat "Vyberte Zdroj" obrazovku pred nacitavanim videa, ale nacitat video linky a vybrat si z nich, s moznostou sa k nim vratit.

## Verzia 0.72 beta 6
**Vseobecne**

* aktualizovana prima-play(prehravanie funkcne)
* pridane zobrazenie gstreamer verzie
* zlepsene prehravanie rtmp streamov pri image s gstreamer-1.0 verziou

test_on_reciever.sh
#!/bin/sh
D=$(pushd $(dirname $0) &> /dev/null; pwd; popd &> /dev/null)
P=${D}/tmp/
PLUGINPATH="/usr/lib/enigma2/python/Plugins/Extensions/archivCZSK"

HOST="192.168.7.105"
USERNAME="root"
PASSWORD="openpli3"

if [ ! -z $1 ];
then
	HOST=$1
fi

if [ ! -z $2 ];
then
	USERNAME=$2
fi
if [ ! -z $3 ];
then
	PASSWORD=$3
fi

echo "connection parameters: $HOST $USERNAME $PASSWORD"

echo "creating archivczsk.tar.gz"
rm ${D}/archivyczsk.tar.gz 2> /dev/null

mkdir -p ${P}${PLUGINPATH}
mkdir -p ${P}${PLUGINPATH}/locale/cs/LC_MESSAGES/
mkdir -p ${P}${PLUGINPATH}/locale/sk/LC_MESSAGES/
mkdir -p ${P}/usr/lib/enigma2/python/Components/Converter/

cp -rp ${D}/build/plugin/src/* ${P}${PLUGINPATH}
cp -p ${D}/build/plugin/src/converter/* ${P}/usr/lib/enigma2/python/Components/Converter/

msgfmt ${D}/build/plugin/po/cs.po -o ${P}${PLUGINPATH}/locale/cs/LC_MESSAGES/archivCZSK.mo
msgfmt ${D}/build/plugin/po/sk.po -o ${P}${PLUGINPATH}/locale/sk/LC_MESSAGES/archivCZSK.mo

#echo "compiling to optimized python bytecode"
#python -O -m compileall ${P} 1> /dev/null


echo "cleanup of unnecessary files"
find ${P} -name "*.po" -exec rm {} \;
find ${P} -name "*.pyo" -exec rm {} \;
find ${P} -name "*.pyc" -exec rm {} \;
find ${P} -name Makefile.am -exec rm {} \;

rm -rf ${P}${PLUGINPATH}/converter
rm -rf ${P}${PLUGINPATH}/engine/player/servicemp4

${D}/build/plugin/src/script/getaddons.py xbmc_doplnky ${P}
${D}/build/plugin/src/script/getaddons.py dmd_czech ${P}
${D}/build/plugin/src/script/getaddons.py custom ${P}

tar -C ${P} -czf archivyczsk.tar.gz .

echo "uploading archivczsk.tar.gz to $HOST"
ftp -n $HOST <<EOFFTP
user $USERNAME $PASSWORD
binary
cd /tmp
put archivyczsk.tar.gz
bye
EOFFTP

rm archivyczsk.tar.gz 2> /dev/null
rm -rf ${P}

echo "installing archivCZSK on $HOST"
sshpass -p $PASSWORD ssh $USERNAME@$HOST << EOFSSH
rm -rf $PLUGINPATH
tar -C / -xzf /tmp/archivyczsk.tar.gz
rm /tmp/archivyczsk.tar.gz
killall enigma2
EOFSSH


echo "archivCZSK was succesfully installed on $HOST"
echo "restarting enigma2"


install.sh
#!/bin/bash

D=$(pushd $(dirname $0) &> /dev/null; pwd; popd &> /dev/null)
PLUGINPATH="/usr/lib/enigma2/python/Plugins/Extensions/archivCZSK"

if [ ! -z $1 ];
then
	E2_HOST=$1
fi

if [ ! -z $2 ];
then
	E2_USERNAME=$2
fi
if [ ! -z $3 ];
then
	E2_PASSWORD=$3
fi

echo "connection parameters: $E2_HOST $E2_USERNAME $E2_PASSWORD"
echo "creating ipk..."
${D}/create_ipk_all.sh > /dev/null
IPK_PATH=$(find ${D} -maxdepth 1 -mmin 1 -name "*.ipk")
IPK_NAME=$(basename $IPK_PATH)

echo "uploading ipk to $E2_HOST..."
ftp -n $E2_HOST <<EOFFTP
user $E2_USERNAME $E2_PASSWORD
binary
cd /tmp
lcd ${D}
put $IPK_NAME
bye
EOFFTP

echo "installing archivCZSK on $E2_HOST"
sshpass -p $E2_PASSWORD ssh -t $E2_USERNAME@$E2_HOST << EOFSSH
opkg remove enigma2-plugin-extensions-archivczsk
opkg install /tmp/$IPK_NAME
rm /tmp/$IPK_NAME
killall enigma2
EOFSSH

echo "restarting enigma2"


release_zip_only.sh
# a='hello'
# b='world'
# c=$a$b
# echo $c

# cez args spravit

if [ -z $1 ];
then
	echo " "
	echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
	echo "!!!! first args must be a version number etc 0.75 !!!!"
	echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
	echo " "
else
	version=$1
	D=$(pushd $(dirname $0) &> /dev/null; pwd; popd &> /dev/null)
	cd $D
	rm -rf archivCZSK/
	sed -i "s/version = \".*\"/version = \"$version\"/g" build/plugin/src/version.py
	cp -a build/plugin/src/. archivCZSK/
	mkdir archivCZSK/locale
	mkdir archivCZSK/locale/sk
	mkdir archivCZSK/locale/cs
	mkdir archivCZSK/locale/cs/LC_MESSAGES
	mkdir archivCZSK/locale/sk/LC_MESSAGES
	msgfmt build/plugin/po/sk.po -o archivCZSK/locale/sk/LC_MESSAGES/archivCZSK.mo
	msgfmt build/plugin/po/cs.po -o archivCZSK/locale/cs/LC_MESSAGES/archivCZSK.mo
	
	zip -FS -q -r build/plugin/update/version/archivczsk-$version.zip archivCZSK -x "*.py[oc] *.sw[onp]" -x "*converter/*" -x "*Makefile.am" -x "*categories.xml" -x "*.gitignore"
	rm -rf archivCZSK/
	cp build/plugin/src/changelog.txt build/plugin/update/version/changelog-$version.txt
	sed -i "s/version=\".*\">/version=\"$version\">/g" build/plugin/update/app.xml
	echo " "
	echo "****************************************************************"
	echo "**** ZIP & changelog create to build/plugin/update/version/ ****"
	echo "**** Please check this file: build/plugin/update/app.xml    ****"
	echo "****                         build/plugin/src/version.py    ****"
	echo "****************************************************************"
	echo " "
fi


create_ipk_twisted.sh
if [ ! "`whoami`" = "root" ]
then
    echo "*************************** Please run script as root ***************************"
    exit 1
fi

echo " "
echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
echo "!!!! run command 'sudo su' then run script !!!!"
echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
echo " "


##################################
############ SETTINGS ############
##################################


############ SCRIPT ############
D=$(pushd $(dirname $0) &> /dev/null; pwd; popd &> /dev/null)
P=${D}/ipkgtwisted.tmp.$$
B=${D}/ipkgtwisted.build.$$
Tw=${D}/build/twisted
PKG=${D}/twisted-16-2-0-fix

pushd ${D} &> /dev/null

popd &> /dev/null

rm -rf ${D}/ipkgtwisted.tmp*
rm -rf ${D}/ipkgtwisted.build*

mkdir -p ${P}
mkdir -p ${P}/CONTROL
mkdir -p ${B}

cat > ${P}/CONTROL/control << EOF
Package: python-twisted
Version: 16.2.0
Architecture: all
Section: extra
Priority: optional
Depends: enigma2-plugin-extensions-archivczsk (>=0.74)
Recommends: python-html, python-threading
Maintainer: ferko
Homepage: https://github.com/mx3L/archivczsk/releases
Description: archivCZSK fix BlackHole image
EOF

cat > ${P}/CONTROL/preinst << EOF
#!/bin/sh
echo "preinst"
exit 0
EOF

cat > ${P}/CONTROL/prerm << EOF
#!/bin/sh
echo "prerm"
exit 0
EOF

cat > ${P}/CONTROL/postinst << EOF
#!/bin/sh
echo "installing twisted"
if [ -d $PSitePackages];
then
	if [ -d /usr/lib/python2.7/site-packages/twisted ];
	then
		echo "twisted found... REINSTALL"
		cp -R /usr/lib/python2.7/site-packages/twisted /usr/lib/python2.7/site-packages/twisted_origin_backup
		echo "create backup to twisted_origin_backup"
		rm -rf /usr/lib/python2.7/site-packages/twisted
		cp -R /tmp/twisted /usr/lib/python2.7/site-packages/twisted
	else
		echo "twisted not found... INSTALL"
		cp -R /tmp/twisted /usr/lib/python2.7/site-packages/twisted
	fi
else
	echo "python 2.7 site-packages dir not found!!!"
fi
echo "Install twisted successfully, please reboot device."
exit 0
EOF

cat > ${P}/CONTROL/postrm << EOF
#!/bin/sh
echo "postrm"
exit 0
EOF

chmod 755 ${P}/CONTROL/preinst
chmod 755 ${P}/CONTROL/postinst
chmod 755 ${P}/CONTROL/prerm
chmod 755 ${P}/CONTROL/postrm

mkdir -p ${P}/tmp

cp -R ${Tw} ${P}/tmp

tar -C ${P} -czf ${B}/data.tar.gz . --exclude=CONTROL
tar -C ${P}/CONTROL -czf ${B}/control.tar.gz .

echo "2.0" > ${B}/debian-binary

cd ${B}
ls -la
ar -r ${PKG}.ipk ./debian-binary ./control.tar.gz ./data.tar.gz
ar -r ${PKG}.deb ./debian-binary ./control.tar.gz ./data.tar.gz
cd ${D}

rm -rf ${P}
rm -rf ${B}

create_ipk_mipsel.sh
# script taken from openwebif project
D=$(pushd $(dirname $0) &> /dev/null; pwd; popd &> /dev/null)
P=${D}/ipkg.tmp.$$
B=${D}/ipkg.build.$$
DP=${D}/ipkg.deps

P26="http://www.python.org/ftp/python/2.6/Python-2.6.tgz"
P27="http://www.python.org/ftp/python/2.7.5/Python-2.7.5.tgz"

SH4_V="0.1"
MIPS_V="0.2"
SH4="http://dl.bintray.com/mx3l/generic/sh4_$SH4_V.tar.gz"
MIPS="http://dl.bintray.com/mx3l/generic/mipsel_$MIPS_V.tar.gz"

pushd ${D} &> /dev/null

PVER="0.70"
GITVER=$(git log -1 --format="%ci" | awk -F" " '{ print $1 }' | tr -d "-")
DSTAGE="beta"
DSTAGEVER="7"
VER=$PVER\_$DSTAGE\_$DSTAGEVER\_$GITVER

PKG=${D}/enigma2-plugin-extensions-archivczsk_${VER}_mipsel.ipk
PLUGINPATH=/usr/lib/enigma2/python/Plugins/Extensions/archivCZSK
popd &> /dev/null

mkdir -p ${P}
mkdir -p ${P}/CONTROL
mkdir -p ${B}
mkdir -p ${DP}

if [ -d ${DP}/Python-2.6 ] && [ -d ${DP}/Python-2.7 ]; then
	echo "python packages are already downloaded"
else
	echo "downloading neccesary python packages..."
	wget -O ${DP}/Python-2.6.tgz $P26
	wget -O ${DP}/Python-2.7.5.tgz $P27
	tar -C ${DP} -xzf ${DP}/Python-2.6.tgz
	tar -C ${DP} -xzf ${DP}/Python-2.7.5.tgz
	mv ${DP}/Python-2.7.5 ${DP}/Python-2.7
fi

if [ -d ${DP}/sh4 ] && [ -d ${DP}/mipsel ]; then
	echo "sh4/mipsel binaries are already downloaded"
else
	echo "downloading neccesary sh4/mipsel binaries"
	wget -O ${DP}/sh4_$SH4_V.tar.gz $SH4
	wget -O ${DP}/mipsel_$MIPS_V.tar.gz $MIPS
	tar -C ${DP} -xzf ${DP}/sh4_$SH4_V.tar.gz
	tar -C ${DP} -xzf ${DP}/mipsel_$MIPS_V.tar.gz
fi


cat > ${P}/CONTROL/control << EOF
Package: enigma2-plugin-extensions-archivczsk
Version: ${VER}
Architecture: mipsel
Section: extra
Priority: optional
Depends: python-html python-threading python-json python-encodings python-zlib
Maintainer: mxfitsat@gmail.com
Homepage: https://code.google.com/p/archivy-czsk/
Source: https://code.google.com/p/archivy-czsk/
Description: prehravanie CZ/SK archivov $VER"
EOF

cat > ${P}/CONTROL/preinst << EOF
#!/bin/sh
rm -r /usr/lib/enigma2/python/Plugins/Extensions/archivCZSK 2> /dev/null
exit 0
EOF

cat > ${P}/CONTROL/prerm << EOF
#!/bin/sh
rm -r /usr/lib/enigma2/python/Plugins/Extensions/archivCZSK 2> /dev/null
exit 0
EOF

cp ${D}/script/postinst ${P}/CONTROL/

chmod 755 ${P}/CONTROL/preinst
chmod 755 ${P}/CONTROL/postinst
chmod 755 ${P}/CONTROL/prerm

mkdir -p ${P}${PLUGINPATH}
mkdir -p ${P}${PLUGINPATH}/locale/cs/LC_MESSAGES/
mkdir -p ${P}${PLUGINPATH}/locale/sk/LC_MESSAGES/
mkdir -p ${P}/usr/lib/enigma2/python/Components/Converter/

cp -rp ${D}/build/plugin/src/* ${P}${PLUGINPATH}
cp -p ${D}/build/plugin/src/converter/* ${P}/usr/lib/enigma2/python/Components/Converter/

echo "creating locales"
msgfmt ${D}/build/plugin/po/cs.po -o ${P}${PLUGINPATH}/locale/cs/LC_MESSAGES/archivCZSK.mo
msgfmt ${D}/build/plugin/po/sk.po -o ${P}${PLUGINPATH}/locale/sk/LC_MESSAGES/archivCZSK.mo

echo "compiling to optimized python bytecode"
python -O -m compileall ${P} 1> /dev/null

echo "cleanup of unnecessary files"
find ${P} -name "*.po" -exec rm {} \;
find ${P} -name "*.py" -exec rm {} \;
find ${P} -name "*.pyc" -exec rm {} \;
find ${P} -name Makefile.am -exec rm {} \;
rm -rf ${P}${PLUGINPATH}/converter
rm -rf ${P}${PLUGINPATH}/engine/player/servicemp4
rm -rf ${P}${PLUGINPATH}/resources/data/*

echo "downloading addons"
${D}/build/plugin/src/script/getaddons.py xbmc_doplnky ${P}
${D}/build/plugin/src/script/getaddons.py dmd_czech ${P}
${D}/build/plugin/src/script/getaddons.py custom ${P}

mkdir -p ${P}/tmp/archivczsk/python2.7
mkdir -p ${P}/tmp/archivczsk/python2.6

cp -rp ${DP}/sh4 ${P}/tmp/archivczsk
cp -rp ${DP}/mipsel ${P}/tmp/archivczsk

cp -p ${D}/script/getenigma2ver ${P}/tmp/archivczsk

cp -p ${DP}/Python-2.6/Lib/encodings/hex_codec.py ${P}/tmp/archivczsk/python2.6/hex_codec.py
cp -p ${DP}/Python-2.6/Lib/encodings/string_escape.py ${P}/tmp/archivczsk/python2.7/string_escape.py
cp -p ${DP}/Python-2.6/Lib/encodings/latin_1.py ${P}/tmp/archivczsk/python2.6/latin_1.py
cp -p ${DP}/Python-2.6/Lib/encodings/utf_16.py ${P}/tmp/archivczsk/python2.6/utf_16.py
cp -p ${DP}/Python-2.6/Lib/encodings/idna.py ${P}/tmp/archivczsk/python2.6/idna.py
cp -p ${DP}/Python-2.6/Lib/encodings/iso8859_2.py ${P}/tmp/archivczsk/python2.6/iso8859_2.py
cp -p ${DP}/Python-2.6/Lib/encodings/cp1250.py ${P}/tmp/archivczsk/python2.6/cp1250.py
cp -p ${DP}/Python-2.6/Lib/decimal.py ${P}/tmp/archivczsk/python2.6/decimal.py
cp -p ${DP}/Python-2.6/Lib/formatter.py ${P}/tmp/archivczsk/python2.6/formatter.py
cp -p ${DP}/Python-2.6/Lib/markupbase.py ${P}/tmp/archivczsk/python2.6/markupbase.py
cp -p ${DP}/Python-2.6/Lib/HTMLParser.py ${P}/tmp/archivczsk/python2.6/HTMLParser.py
cp -p ${DP}/Python-2.6/Lib/htmlentitydefs.py ${P}/tmp/archivczsk/python2.6/htmlentitydefs.py
cp -p ${DP}/Python-2.6/Lib/htmllib.py ${P}/tmp/archivczsk/python2.6/htmllib.py
cp -p ${DP}/Python-2.6/Lib/sgmllib.py ${P}/tmp/archivczsk/python2.6/sgmllib.py
cp -p ${DP}/Python-2.6/Lib/stringprep.py ${P}/tmp/archivczsk/python2.6/stringprep.py
cp -p ${DP}/Python-2.6/Lib/numbers.py ${P}/tmp/archivczsk/python2.6/numbers.py
cp -p ${DP}/Python-2.6/Lib/subprocess.py ${P}/tmp/archivczsk/python2.6/subprocess.py
cp -p ${DP}/Python-2.6/Lib/_LWPCookieJar.py ${P}/tmp/archivczsk/python2.6/_LWPCookieJar.py
cp -p ${DP}/Python-2.6/Lib/_MozillaCookieJar.py ${P}/tmp/archivczsk/python2.6/_MozillaCookieJar.py
cp -p ${DP}/Python-2.6/Lib/cookielib.py ${P}/tmp/archivczsk/python2.6/cookielib.py
cp -p ${DP}/Python-2.6/Lib/shutil.py ${P}/tmp/archivczsk/python2.6/shutil.py
cp -p ${DP}/Python-2.6/Lib/fnmatch.py ${P}/tmp/archivczsk/python2.6/fnmatch.py
cp -p ${DP}/Python-2.6/Lib/threading.py ${P}/tmp/archivczsk/python2.6/threading.py
cp -p ${DP}/Python-2.6/Lib/zipfile.py ${P}/tmp/archivczsk/python2.6/zipfile.py
cp -p ${DP}/Python-2.6/Lib/httplib.py ${P}/tmp/archivczsk/python2.6/httplib.py
cp -p ${DP}/Python-2.6/Lib/stat.py ${P}/tmp/archivczsk/python2.6/stat.py

cp -p ${DP}/Python-2.7/Lib/encodings/hex_codec.py ${P}/tmp/archivczsk/python2.7/hex_codec.py
cp -p ${DP}/Python-2.7/Lib/encodings/string_escape.py ${P}/tmp/archivczsk/python2.7/string_escape.py
cp -p ${DP}/Python-2.7/Lib/encodings/latin_1.py ${P}/tmp/archivczsk/python2.7/latin_1.py
cp -p ${DP}/Python-2.7/Lib/encodings/utf_16.py ${P}/tmp/archivczsk/python2.7/utf_16.py
cp -p ${DP}/Python-2.7/Lib/encodings/idna.py ${P}/tmp/archivczsk/python2.7/idna.py
cp -p ${DP}/Python-2.7/Lib/encodings/iso8859_2.py ${P}/tmp/archivczsk/python2.7/iso8859_2.py
cp -p ${DP}/Python-2.7/Lib/encodings/cp1250.py ${P}/tmp/archivczsk/python2.7/cp1250.py
cp -p ${DP}/Python-2.7/Lib/decimal.py ${P}/tmp/archivczsk/python2.7/decimal.py
cp -p ${DP}/Python-2.7/Lib/formatter.py ${P}/tmp/archivczsk/python2.7/formatter.py
cp -p ${DP}/Python-2.7/Lib/markupbase.py ${P}/tmp/archivczsk/python2.7/markupbase.py
cp -p ${DP}/Python-2.7/Lib/HTMLParser.py ${P}/tmp/archivczsk/python2.7/HTMLParser.py
cp -p ${DP}/Python-2.7/Lib/htmlentitydefs.py ${P}/tmp/archivczsk/python2.7/htmlentitydefs.py
cp -p ${DP}/Python-2.7/Lib/htmllib.py ${P}/tmp/archivczsk/python2.7/htmllib.py
cp -p ${DP}/Python-2.7/Lib/sgmllib.py ${P}/tmp/archivczsk/python2.7/sgmllib.py
cp -p ${DP}/Python-2.7/Lib/stringprep.py ${P}/tmp/archivczsk/python2.7/stringprep.py
cp -p ${DP}/Python-2.7/Lib/numbers.py ${P}/tmp/archivczsk/python2.7/numbers.py
cp -p ${DP}/Python-2.7/Lib/subprocess.py ${P}/tmp/archivczsk/python2.7/subprocess.py
cp -p ${DP}/Python-2.7/Lib/_LWPCookieJar.py ${P}/tmp/archivczsk/python2.7/_LWPCookieJar.py
cp -p ${DP}/Python-2.7/Lib/_MozillaCookieJar.py ${P}/tmp/archivczsk/python2.7/_MozillaCookieJar.py
cp -p ${DP}/Python-2.7/Lib/cookielib.py ${P}/tmp/archivczsk/python2.7/cookielib.py
cp -p ${DP}/Python-2.7/Lib/shutil.py ${P}/tmp/archivczsk/python2.7/shutil.py
cp -p ${DP}/Python-2.7/Lib/fnmatch.py ${P}/tmp/archivczsk/python2.7/fnmatch.py
cp -p ${DP}/Python-2.7/Lib/threading.py ${P}/tmp/archivczsk/python2.7/threading.py
cp -p ${DP}/Python-2.7/Lib/zipfile.py ${P}/tmp/archivczsk/python2.7/zipfile.py
cp -p ${DP}/Python-2.7/Lib/httplib.py ${P}/tmp/archivczsk/python2.7/httplib.py
cp -p ${DP}/Python-2.7/Lib/stat.py ${P}/tmp/archivczsk/python2.7/stat.py


tar -C ${P} -czf ${B}/data.tar.gz . --exclude=CONTROL
tar -C ${P}/CONTROL -czf ${B}/control.tar.gz .

echo "2.0" > ${B}/debian-binary

cd ${B}
ls -la
ar -r ${PKG} ./debian-binary ./data.tar.gz ./control.tar.gz
cd -

rm -rf ${P}
rm -rf ${B}
